### DISCUSSION TITLE
Filtering TreeState items

### DISCUSSION THREAD

**Discussion by Joonaspraks** [adobe/react-spectrum#5578] [Q&A] [answered]
Hi!

I am exploring the SubMenu hooks on the nightly build and wanted to throw in a search bar to filter the available actions on the menu. Sadly, I found that `useTreeState` has no built-in filtering (like `useListState` does). Is it by design? Because it's not clear whether the user would like to filter only the top level nodes or child nodes as well?

**Comment by reidbarber** [comment, marked as answer]
I think this is something we want, but just haven't implemented yet. And I think it would filter all nodes, where the implementer could account for level or expandedKeys if they wanted to in their filter function.

**Reply by snowystinger** [reply]
Something to keep in mind, search fields inside a menu are not valid aria. You'll need to put the search outside or build sub dialogs instead of sub menus. We don't quite support that yet though, but it's on the radar.
